ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/ContractsReader.scala

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events
+
+import java.time.Instant
+
+import anorm.SqlParser.{binaryStream, int, str}
+import anorm.{Row, RowParser, SimpleSql, SqlStringInterpolation, ~}
+import com.daml.ledger.participant.state.index.v2.ContractStore
+import com.daml.platform.store.Conversions._
+import com.daml.platform.store.{DbType, dao}
+import com.daml.platform.store.dao.{DbDispatcher, events}
+import com.daml.platform.store.serialization.KeyHasher.{hashKey => hash}
+import com.daml.platform.store.serialization.ValueSerializer.{deserializeValue => deserialize}
+
+import scala.concurrent.Future
+
+/**
+  * @see [[ContractsTable]]
+  */
+private[dao] sealed abstract class ContractsReader(dispatcher: DbDispatcher) extends ContractStore {
+
+  import ContractsReader._
+
+  protected def lookupContractKeyQuery(submitter: Party, key: dao.events.Key): SimpleSql[Row]
+
+  override def lookupActiveContract(
+      submitter: Party,
+      contractId: ContractId,
+  ): Future[Option[Contract]] =
+    dispatcher.executeSql("lookup_active_contract") { implicit connection =>
+      SQL"select participant_contracts.contract_id, template_id, create_argument from #$contractsTable where contract_witness = $submitter and participant_contracts.contract_id = $contractId"
```

Comment:
To make the query a bit shorter and not spell out the join twice.

Reply:
Hm, I'm not sure the brevity is worth it tbh. I'd rather see a full page of query text rather than having to do mental dictionary lookups to the side.
But if this is the pattern we've been following so far (and I suspect it is), then let's continue with it.